src/nvim/os/fs.c

```diff
@@ -114,17 +114,19 @@ bool os_can_exe(const char_u *name, char_u **abspath)
 static bool is_executable(const char_u *name)
   FUNC_ATTR_NONNULL_ALL
 {
+#ifdef WIN32
+    // Windows does not have an executable bit.
+    // Instead check if the file exists and is readable.
+    return os_file_is_readable((char *) name);
+#else
```

Comment:
I think the logic is ok, can you give a scenario where it would be wrong? The call to `GetFileAttributes()` above that will fail if the file doesn't exist.